docs/articles/scaleout.md

```diff
@@ -0,0 +1,154 @@
+# Easy scaling of a zot cluster
+
+> :point_right: A cluster of zot replicas can easily be scaled with no repo-specific programming of the load balancer using:
+>
+> -   Stateless zot instances to simplify scale out
+> -   Shared remote storage
+> -   zot release v2.1.0 or later
+
+Beginning with zot release v2.1.0, a new "scale-out" architecture greatly reduces the configuration required when deploying large numbers of zot instances.  As before, multiple identical zot replicas run simultaneously using the same shared reliable storage, but with improved scale and performance in large deployments. 
+
+![504569](../assets/images/504569.jpg){width="500"}
+
+The number of replicas can easily be expanded by simply adding the IP addresses of the new replicas in the load balancer configuration. No repo-specific programming of the load balancer is needed. The shared storage can also be easily increased or decreased.
+
+> :pencil2: For high availability clustering with earlier zot releases, see [zot Clustering](clustering.md).
+
+## Prerequisites
+
+For easy scaling of replicas, the following conditions must be met:
+
+- All zot replicas must be running zot release v2.1.0 (or later) with identical configurations.
+- All zot replicas in the cluster use remote storage at a single shared S3 backend. There is no local caching in the zot replicas.
+- Each repo is served by one zot replica, and that replica is solely responsible for serving all images of that repo. 
+- A repo in storage can be written to only by the zot replica associated with that repo.
+- Each zot replica in the cluster has its own IP address, but all replicas use the port number.
+- The URI format sent to the load balancer must be /v2/<repo\>/<manifest\>:<tag\>
+
+Beginning with zot release v2.1.0, garbage collection is allowed in the shared cluster storage.
+
+## How it works
+
+A highly scalable cluster can be architected by sharding on the repository name. In the cluster, each replica is the owner of a small subset of the repository. The load balancer does not need to know which replica owns which repo. The replicas themselves can determine this.  
+
+When the load balancer receives an image push or pull request, it forwards the request to any replica in the cluster. The receiving replica hashes the repo path and consults a hash table in shared storage to determine which replica is responsible for the repo. The receiving replica forwards the request to the responsible replica and then acts as a proxy, returning the requested image to the requestor.
```

Comment:
"in shared storage" ... drop this

Reply:
The hash lookup determines if the request needs to be handled locally or forwarded to the right zot instance.

Reply:
Note that we use siphash as our hashing algorithm for better collision and pre-image resistance.

Reply:
Add this somewhere here ... note that the zot instances in the cluster can be exposed directly to clients as well without the need for a load balancer. For example, DNS based routing.